### ISSUE TITLE
`unused_async` false positive when await statement is behind a feature flag

### ISSUE THREAD

**Issue by m4tx** [rust-lang/rust-clippy#15885]
### Summary

Clippy raises an `unused_async` lint error when the `await` statement is present, but gated behind a feature flag. It acts as if only the default features are enabled and ignores everything else.

### Lint Name

unused_async

### Reproducer

I tried this code:

```rust
#![deny(clippy::unused_async)]

async fn test() {
    #[cfg(feature = "myflag")]
    foo().await
}

async fn foo() {
    todo!()
}
```

I saw this happen:

```
error: unused `async` for function with no await statements
  --> src/main.rs:7:1
   |
 7 | / async fn test() {
 8 | |     #[cfg(feature = "test")]
 9 | |     foo().await
10 | | }
   | |_^
   |
   = help: consider removing the `async` from this function
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_async
```

When the `myflag` feature flag is included in the `default` feature flag set for the crate, the error is gone.

### Version

```text
rustc 1.92.0-nightly (4b94758d2 2025-10-13)
binary: rustc
commit-hash: 4b94758d2ba7d0ef71ccf5fde29ce4bc5d6fe2a4
commit-date: 2025-10-13
host: x86_64-unknown-linux-gnu
release: 1.92.0-nightly
LLVM version: 21.1.3
```
